### DISCUSSION TITLE
Generate queryRaw from user input based on variable

### DISCUSSION THREAD

**Discussion by 404answernotfound** [prisma/prisma#13454] [Q&A] [answered]
Hello, I wanted to use queryRaw to query the database with a built statement. We have a query builder that would pass a query to queryRaw and access the database, but it's not working (maybe that's intended?)

Here is some example code:

```javascript

            if(input){

                console.log(input) // This is the value of Input -> SELECT * FROM car WHERE car."plate" ILIKE '%123%' of type String
                const input = '%123%'

                // const result =  prisma.$queryRaw`SELECT * FROM car WHERE car."plate" ILIKE '%123%'` // works

                // const result =  prisma.$queryRaw`SELECT * FROM car WHERE car."plate" ILIKE ${param}` // works

                // const result =  prisma.$queryRawUnsafe(input) // works

                const result =  prisma.$queryRaw`${input}` // Doesn't work

                return result

            }

         // ... Other code
        }    
```

Is there a way to avoid using queryRawUnsafe while still sending a prepared query that was built by the builder on our backend?

Thanks!

**Comment by nurul3101** [comment, marked as answer]
Hey @404answernotfound 👋 

`queryRaw` method is implemented as a [tagged template](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates), which allows you to pass a template literal where you can easily insert your variables. Template variables can only be used for data values and cannot be used for identifiers such as column names, table names or database names, or for SQL keywords.
It seems you are trying to pass an entire query in a single variable due to which you are getting an error: [QueryRaw Reference](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access)

**Reply by 404answernotfound** [reply]
So if we want to pass an entire query which we are building dynamically we should use rawQueryUnsafe, correct? What about sanitization of the query? Does Prisma sanitize at a basic level with rawqueryunsafe or its entirely in the consumer hands to build the query safely?

**Reply by nurul3101** [reply]
Yes `queryRawUnsafe` could be used in this case, Using queryRawUnsafe opens up the possibility for SQL injection attacks so you should sanitize the query. It's upon the consumer to sanitize the input.

**Reply by 404answernotfound** [reply]
Thank you @nurul3101

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
